Pal 1056 tos documentation

developer01189998819991197253 opened the pull request.

pd104923 commented on topic-offset-service/README.md:
```diff
@@ -0,0 +1,92 @@
+
+This service accepts a `TopicOffsetRequest`, which contains a token, it is then used to subscribe to the akka stream and check for a Start of Stream message, declared in the `StreamMarker` class. 
+When a Start of Stream message is observed, the commitOffset is retrieved, and sent onward to the [Filtered-Resource Service](../filtered-resource-service) to point the 
+Filtered-Resource Service to the commitOffset on the Kafka queue.
```
You've mentioned that it subscribes to an akka stream to get the start of stream offset value. I thought this was done on the `masked-resource` kafka topic.
I would also update the last part to refer to a kafka topic rather than a queue.

pd104923 commented on topic-offset-service/src/component-tests/java/uk/gov/gchq/palisade/component/topicoffset/model/TopicOffsetResponseTest.java:
```diff
@@ -15,59 +15,42 @@
- * Tests for the evaluating the TopicOffsetResponse and the related serialising to a JSON string
- * and deseralising back to an object.
+ * Tests for the evaluating the TopicOffsetResponse, and the related serialising to a JSON string
+ * and deserialising back to an object.
```
```suggestion
 * Tests for evaluating the TopicOffsetResponse, and the related serialising to a JSON string
 * and deserialising back to an object.
```

pd104923 commented on topic-offset-service/src/component-tests/java/uk/gov/gchq/palisade/component/topicoffset/web/TopicOffsetRestControllerTest.java:
```diff
@@ -88,10 +88,11 @@ void testContextLoads() {
     /**
      * This permutation is for a request that comes in for an end of messages indicator, StreamMarker is set to START.
```
Request for an end of messages indicator? Is this correct when the StreamMarker is set to `START`?

pd104923 commented on topic-offset-service/src/component-tests/java/uk/gov/gchq/palisade/component/topicoffset/model/TopicOffsetResponseTest.java:
```diff
@@ -15,59 +15,42 @@
      *
-     * @throws IOException throws if the {@link TopicOffsetResponse} object cannot be converted to a JsonContent.
-     *                     This equates to a failure to serialise or deserialise the string.
+     * @throws JsonProcessingException throws if the {@link TopicOffsetResponse} object cannot be converted to a JsonContent.
```
```suggestion
     * @throws JsonProcessingException if the {@link TopicOffsetResponse} object cannot be converted to a JsonContent.
```

pd104923 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/TopicOffsetApplication.java:
```diff
@@ -38,10 +38,10 @@
  * be watching for a Kafka header with the message {Stream-Marker=Start, Token=xxxx-xxxx-xxxx}. It will take this
- * information along with the commit offset of this stream and this will be written to the downstream queue.  This can
+ * information along with the commit offset of this stream and this will be written to the downstream queue. This can
  * then be used to optimise the start up client connections by the filtered-resource-service.
```
```suggestion
 * then be used to optimise the start up client connections by the Filtered-Resource Service.
```

pd104923 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/model/TopicOffsetResponse.java:
```diff
@@ -26,15 +26,15 @@
 /**
- * The TopicOffsetResponse is the output for topic-offset-service. It will only be created if the request message was
- * found to be the start of the messages for a specific client request. It will will contain the offset in the message
+ * The TopicOffsetResponse is the output for Topic-Offset Service. It will only be created if the request message was
+ * found to be the start of the messages for a specific client request. It will contain the offset in the message
  * queue for this first message. This is forwarded to the masked-resource-offset stream where it is used to prepare for
  * the client request to retrieve the data.
```
```suggestion
 * The TopicOffsetResponse is the output for Topic-Offset Service. It will only be created if the request message was
 * found to be the start marker for the messages of a specific client request. It will contain the offset in the message
 * queue for this first message. This is forwarded to the masked-resource-offset Kafka topic where it is used to prepare for
 * the client request to retrieve the data.
```

pd104923 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/service/KafkaHealthIndicator.java:
```diff
@@ -36,7 +36,7 @@
 
 /**
- * Kafka health indicator. Check that the consumer group can be accessed and is registered with the cluster,
+ * Kafka health indicator. Check e consumer group can be accessed and is registered with the cluster,
```
This should be reverted back to what it was.

pd104923 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/stream/StreamComponents.java:
```diff
@@ -49,36 +49,36 @@
-     * @param system            the application's actor system to use to load config values
-     * @param keyDeserializer   the stream's key deserialiser
-     * @param valueDeserializer the stream's value deserialiser
+     * @param system            the application's actor system to used to load config values
```
```suggestion
     * @param system            the application's actor system used to load config values
```

pd104923 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/stream/StreamComponents.java:
```diff
@@ -49,36 +49,36 @@
      * Construct an Akka Kafka CommitterSettings from the given config
      *
-     * @param system the application's actor system to use to load config values
+     * @param system the application's actor system to used to load config values
```
```suggestion
     * @param system the application's actor system used to load config values
```

pd104923 reviewed (changes requested):
A few questions and suggestions

pd104923 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/stream/StreamComponents.java:
```diff
@@ -49,36 +49,36 @@
-     * @param system          the application's actor system to use to load config values
-     * @param keySerializer   the stream's key serialiser
-     * @param valueSerializer the stream's value serialiser
+     * @param system          the application's actor system to used to load config values
```
```suggestion
     * @param system          the application's actor system used to load config values
```

pd104923 reviewed (changes requested):
One minor change

dev930018 commented on topic-offset-service/README.md:
```diff
@@ -0,0 +1,91 @@
+# <img src="../logos/logo.svg" width="180">
+
+# Topic Offset Service
+
```
Opening sentence along the lines of "Watches kafka topis and reports offsets for messages that satisfy some prerequisite"

dev930018 commented on topic-offset-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/topicoffset/ContractTestData.java:
```diff
@@ -47,11 +45,8 @@ private ContractTestData() {
-    public static final String RESOURCE_ID = "/test/resourceId";
-    public static final String PURPOSE = "test-purpose";
-    public static final Context CONTEXT = new Context().purpose(PURPOSE);
+    private static final String MASKED_RESOURCE = "masked-resource";
```
Is it worth specifying this is `MASKED_RESOURCE_TOPICNAME` or something?

dev930018 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/model/TopicOffsetRequest.java:
```diff
@@ -33,13 +33,13 @@
- * <li> {@code uk.gov.gchq.palisade.service.attributemask.message.AttributeMaskingResponse} is the output from the attribute-masking-service.
- * <li> {@code uk.gov.gchq.palisade.service.topicoffset.message.TopicOffsetRequest} is the input for this the topic-offset-service.
- * <li> {@code uk.gov.gchq.palisade.service.filteredresource.message.FilteredResourceRequest} is the input for this the filtered-resource-service.
+ * <li> uk.gov.gchq.palisade.service.attributemask.model.AttributeMaskingResponse is the output from the Attribute-Masking Service.
+ * <li> {@link TopicOffsetRequest} is the input for this the Topic-Offset Service.
+ * <li> uk.gov.gchq.palisade.service.filteredresource.model.FilteredResourceRequest is the input for this the Filtered-Resource Service.
```
Keep the `@code` formatting.
The last point is irrelevant in this context.
```suggestion
 * <li> {@code uk.gov.gchq.palisade.service.attributemask.model.AttributeMaskingResponse} is the output from the Attribute-Masking Service.
 * <li> {@link TopicOffsetRequest} is the input for this the Topic-Offset Service.
```

dev930018 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/model/TopicOffsetResponse.java:
```diff
@@ -26,15 +26,15 @@
+ * The TopicOffsetResponse is the output for Topic-Offset Service. It will only be created if the request message was
+ * found to be the start marker for the messages of a specific client request. It will contain the offset in the message
+ * queue for this first message. This is forwarded to the masked-resource-offset Kafka topic where it is used to prepare for
  * the client request to retrieve the data.
  */
```
This could do with a similar "this class here is equivalent to that class there" for
```
 * <li> {@code uk.gov.gchq.palisade.service.filteredresource.model.TopicOffsetMessage} is the input for this the Filtered-Resource Service.
```

dev930018 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/service/SimpleTopicOffsetService.java:
```diff
@@ -40,10 +39,10 @@
+        var header = headers.lastHeader(StreamMarker.HEADER);
+        boolean isOffset = ((header != null) && (new String(header.value(), Charset.defaultCharset()).equals(StreamMarker.START.toString())));
         if (isOffset) {
-            LOGGER.info("Found topic offset: {}", headers);
+            LOGGER.debug("Found topic offset: {}", headers);
         }
```
Keep at info, we only see one of these messages per request (like palisade, user service)

dev930018 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/stream/config/AkkaRunnableGraph.java:
```diff
@@ -38,19 +38,19 @@
 
+import uk.gov.gchq.palisade.service.topicoffset.TopicOffsetApplication;
 import uk.gov.gchq.palisade.service.topicoffset.model.TopicOffsetRequest;
 import uk.gov.gchq.palisade.service.topicoffset.model.TopicOffsetResponse;
 import uk.gov.gchq.palisade.service.topicoffset.service.KafkaProducerService;
 import uk.gov.gchq.palisade.service.topicoffset.service.TopicOffsetService;
 import uk.gov.gchq.palisade.service.topicoffset.stream.ConsumerTopicConfiguration;
 import uk.gov.gchq.palisade.service.topicoffset.stream.ProducerTopicConfiguration;
-import uk.gov.gchq.palisade.service.topicoffset.stream.ProducerTopicConfiguration.Topic;
 
 import java.util.Optional;
 import java.util.concurrent.CompletionStage;
 
 /**
- * Configuration for the Akka Runnable Graph used by the {@link uk.gov.gchq.palisade.service.topicoffset.TopicOffsetApplication}
+ * Configuration for the Akka Runnable Graph used by the {@link TopicOffsetApplication}
```
Don't import the `TopicOffsetApplication` just to reference it in javadocs

dev930018 commented on topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/model/TopicOffsetResponse.java:
```diff
@@ -26,15 +26,17 @@
+ * found to be the start marker for the messages of a specific client request. It will contain the offset in the message
+ * queue for this first message. This is forwarded to the masked-resource-offset Kafka topic where it is used to prepare for
  * the client request to retrieve the data.
+ * Note there is another class that effectively represent the same data but represent a different stage of the process.
+ * <li> {@code uk.gov.gchq.palisade.service.filteredresource.model.TopicOffsetMessage} is the input for this the Filtered-Resource Service.
  */
```
Needs `<ul>` tag, and `@link` to this class for consistency with other class

developer01189998819991197253 committed:
PAL-1056 Added Javadoc to the Topic Offset Service
Added a readme with diagram describing the service
Added tos/doc/
Updated javadoc within topic offset service classes, e.g. TopicOffsetRequest.java
Removed AuditErrorMessage.java and AuditErrorMessageTest.java and refactored some javadoc to remove references to auditing
Updated tests so assertion messages are detailed, and simplified some of the test classes.
